### DISCUSSION TITLE
No out-of-box TLS support for nginx ?

### DISCUSSION THREAD

**Discussion by stucky101** [librenms/docker#369] [Q&A]
Hi
I'm trying librenms but I see no mention of TLS anywhere. When poking around I see that people use reverse proxies and I tried that with HAproxy but it totally breaks the GUI since all self-referenced urls are still over http.

I have seen the env var APP_URL pointed to an https link twice in this section but it is not mentioned in the official environment variable list. Anyway I tried it but it has no effect at all. The site looks terrible and all links are still over plain http. Docker inspect shows that APP_URL is def accepted but it seems to have no effect.

"OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "APP_URL=https://mylibretest.domain.com",
                "TZ=America/Los_Angeles",
                "PUID=5000",
                "PGID=5000"

Perhaps the doc is outdated and there are a few env vars I can pass directly to nginx ?

thx

**Comment by crazy-max** [comment]
@stucky101 

> No out-of-box TLS support for nginx ?

There is a [compose example using Traefik](https://github.com/librenms/docker/tree/master/examples/traefik) as a reverse proxy with TLS support.

> I have seen the env var APP_URL pointed to an https link

`APP_URL` is populated through the [`LIBRENMS_BASE_URL` env var](https://github.com/librenms/docker/blob/master/doc/docker/environment-variables.md#misc). It should be the context and not the full URL (default to `/`).

**Comment by stucky101** [comment]
I see this is for Traefik v1x only. I understand that v2 is a complete rewrite. Would someone have a v2 functioning version of this compose file ?

thx

**Comment by crazy-max** [comment]
@stucky101 See #104
